pkg/make/builder.go

```diff
@@ -598,7 +598,19 @@ func (b *Builder) BuildCmd(src, appName string) func(context.Context) error {
 			return err
 		}
 
-		if stderr.Len() > 0 {
+		// With the Sonoma-era Xcode binaries, we've started seeing a bunch of spurious warnings. They appear to
+		// _mostly_ effect our developer build process. In the interest in not breaking the build, we ignore them.
+		// https://github.com/golang/go/issues/62597#issuecomment-1733893918 is some of them, and some seem related to the zig cross compiling
+		stderrStr := stderr.String()
+		if os.Getenv("GITHUB_ACTIONS") != "" {
+			stderrStr = strings.ReplaceAll(stderrStr, "ld: warning: ignoring duplicate libraries: '-lobjc'\n", "")
+			stderrStr = strings.ReplaceAll(stderrStr, "# github.com/kolide/launcher/cmd/launcher\n", "")
+
+			re := regexp.MustCompile(`ld: warning: object file \(.*\) was built for newer 'macOS' version \(14.0\) than being linked \(11.0\)\n`)
```

Comment:
do we want to make this one any more flexible with the version numbers? just wondering if it will still work for those who haven't upgraded yet

Reply:
🤷 We could. Though I hope it's fixed by then.

Reply:
same 🙏 I just wasn't sure if we were seeing 13.0 before this, I thought mine started earlier than the upgrade

Reply:
I still get the error sometimes ad I'm on Ventura still, I don't know how to repro it